skywalking-collector/skywalking-collector-worker/src/main/java/com/a/eye/skywalking/collector/worker/storage/MergePersistenceData.java

```diff
@@ -1,51 +1,25 @@
 package com.a.eye.skywalking.collector.worker.storage;
 
-import java.util.HashMap;
-import java.util.Iterator;
-import java.util.Map;
-import java.util.Spliterator;
-import java.util.function.Consumer;
-
 /**
  * @author pengys5
  */
-public class MergePersistenceData implements Iterable {
+public class MergePersistenceData extends Window<MergeData> implements PersistenceData<MergeData> {
 
-    private Map<String, MergeData> persistenceData = new HashMap<>();
+    private WindowData<MergeData> lockedWindowData;
 
     public MergeData getElseCreate(String id) {
-        if (!persistenceData.containsKey(id)) {
-            persistenceData.put(id, new MergeData(id));
+        if (!lockedWindowData.containsKey(id)) {
+            lockedWindowData.put(id, new MergeData(id));
         }
-        return persistenceData.get(id);
-    }
-
-    public int size() {
-        return persistenceData.size();
-    }
-
-    public void clear() {
-        persistenceData.clear();
-    }
-
-    public MergeData pushOne() {
-        MergeData one = persistenceData.entrySet().iterator().next().getValue();
-        persistenceData.remove(one.getId());
-        return one;
-    }
-
-    @Override
-    public void forEach(Consumer action) {
-        throw new UnsupportedOperationException("forEach");
+        return lockedWindowData.get(id);
     }
 
-    @Override
-    public Spliterator spliterator() {
-        throw new UnsupportedOperationException("spliterator");
+    public void holdData() {
+        lockedWindowData = getCurrentAndHold();
     }
 
-    @Override
-    public Iterator<Map.Entry<String, MergeData>> iterator() {
-        return persistenceData.entrySet().iterator();
+    public void releaseData() {
```

Comment:
`reset` may be better than `releaseData`. First, this is a data class, named `MergePersistenceData`, no need add `Data` as part of method name. Then, `release` is talking something about resources, in here, no resource.

So, as a data window, I prefer `reset`.